### ISSUE TITLE
Failed to create proper link with `ln`

### ISSUE THREAD

**Issue by RocketMaDev** [msys2/Cygwin#132]
I'm using a msys2 distribution bundled with Git, that is the Git Bash (Git 2.39.0). I tried `ln` command, but to my disappointment, the function is kind of weird:

| arg | theoretical | real |
|-----|-------------|-----|
| -s | symlink | copy |
| -L | symlink | hardlink |
| \ | hardlink | hardlink |

Now I use `mklink` bundled with cmd, which works fine. What's the solution?

**Comment by Biswa96** [comment]
You have to use environment variable to change the symlink creation behavior. This page may help https://www.cygwin.com/cygwin-ug-net/using-cygwinenv.html. Here is an example command,

```
MSYS=winsymlinks:native ln -s foo bar
```

**Comment by RocketMaDev** [comment]
OK, I will try it later. But what's the default option? Why it has such a weird behavior?

**Comment by Biswa96** [comment]
> Why it has such a weird behavior?

tl;dr Microsoft. Little bit of history https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/

**Comment by RocketMaDev** [comment]
I've set it, finding a new problem: if I run with privilege, the link it generated is a native symbolic link; however, if I run as a user, the link is valid in msys but an empty file on native Windows.

**Comment by Biswa96** [comment]
You can use `MSYS=winsymlinks:nativestrict ln -s foo bar` to get some error if something is going wrong. Also make sure Developer mode is enabled. See the above link for more details.
